src/test/java/codeu/model/store/basic/MessageStoreTest.java

```diff
@@ -0,0 +1,89 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.Message;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+
+public class MessageStoreTest {
+
+  private MessageStore messageStore;
+  private PersistentStorageAgent mockPersistentStorageAgent;
+
+  private final UUID CONVERSATION_ID_ONE = UUID.randomUUID();
+  private final Message MESSAGE_ONE =
+      new Message(
+          UUID.randomUUID(),
+          CONVERSATION_ID_ONE,
+          UUID.randomUUID(),
+          "message one",
+          Instant.ofEpochMilli(1000));
+  private final Message MESSAGE_TWO =
+      new Message(
+          UUID.randomUUID(),
+          CONVERSATION_ID_ONE,
+          UUID.randomUUID(),
+          "message two",
+          Instant.ofEpochMilli(2000));
+  private final Message MESSAGE_THREE =
+      new Message(
+          UUID.randomUUID(),
+          UUID.randomUUID(),
+          UUID.randomUUID(),
+          "message three",
+          Instant.ofEpochMilli(3000));
+
+  @Before
+  public void setup() {
+    mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+    messageStore = MessageStore.getTestInstance(mockPersistentStorageAgent);
+
+    final List<Message> messageList = new ArrayList<>();
+    messageList.add(MESSAGE_ONE);
+    messageList.add(MESSAGE_TWO);
+    messageList.add(MESSAGE_THREE);
+    messageStore.setMessages(messageList);
+  }
+
+  @Test
+  public void testGetMessagesInConversation() {
+    List<Message> resultMessages = messageStore.getMessagesInConversation(CONVERSATION_ID_ONE);
+
+    Assert.assertEquals(2, resultMessages.size());
+    assertEquals(MESSAGE_ONE, resultMessages.get(0));
+    assertEquals(MESSAGE_TWO, resultMessages.get(1));
+  }
+
+  @Test
+  public void testAddMessage() {
+    UUID inputConversationId = UUID.randomUUID();
+    Message inputMessage =
+        new Message(
+            UUID.randomUUID(),
+            inputConversationId,
+            UUID.randomUUID(),
+            "test message",
+            Instant.now());
+
+    messageStore.addMessage(inputMessage);
+    Message resulMessage = messageStore.getMessagesInConversation(inputConversationId).get(0);
+
+    assertEquals(inputMessage, resulMessage);
+    Mockito.verify(mockPersistentStorageAgent).writeThrough(inputMessage);
+  }
+
+  private void assertEquals(Message expectedMessage, Message actualMessage) {
+    Assert.assertEquals(expectedMessage.getId(), actualMessage.getId());
```

Comment:
This pattern is somewhat suspicious, unless you have other tests to verify the correctness of your getters. A flaw in getId() (for example), could mean that you are not actually comparing the IDs of the two messages.

Reply:
There are tests for the getters in UserTest.

Honestly I'm using this pattern because I didn't want to implement `equals()` and `hashcode()` functions, because that sounds like an interesting problem for students to solve. Although I suppose the same argument could be made for the `equals()` function.

Reply:
OK